craft_parts/packages/__init__.py

```diff
@@ -34,6 +34,11 @@ def _get_repository_for_platform() -> Type["BaseRepository"]:
 
         return Ubuntu
 
+    if is_rpm_based():
+        from .rpm import RPMRepository
+
+        return RPMRepository
```

Comment:
I would change this to:

```suggestion
    if is_yum_based():
        from .yum import YUMRepository

        return YUMRepository
```